common/src/node.rs

```diff
@@ -167,3 +446,264 @@ fn populate_node(graph: &mut Graph, node: NodeId) {
         chunks: Chunks::default(),
     });
 }
+
+/// Represents a particular axis in a voxel grid.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum CoordAxis {
+    X = 0,
+    Y = 1,
+    Z = 2,
+}
+
+impl CoordAxis {
+    /// Iterates through the the axes in ascending order
+    pub fn iter() -> impl ExactSizeIterator<Item = Self> {
+        [Self::X, Self::Y, Self::Z].iter().copied()
+    }
+
+    /// Returns the pair axes orthogonal to the current axis
+    pub fn other_axes(self) -> [Self; 2] {
+        match self {
+            Self::X => [Self::Y, Self::Z],
+            Self::Y => [Self::Z, Self::X],
+            Self::Z => [Self::X, Self::Y],
+        }
+    }
+}
+
+impl TryFrom<usize> for CoordAxis {
+    type Error = anyhow::Error;
```

Comment:
`anyhow::Error` is a poor choice for trivial errors not intended to be user-facing, as it's a non-zero-sized, non-`Copy` type. Good style is to define an appropriate error type, or just use a `fn new` that returns `Option` when it's self-evident (compare [`NonZeroU32::new`](https://doc.rust-lang.org/std/num/struct.NonZeroU32.html#method.new).

Reply:
Ah, good to know. I'll go ahead with `Option`, as that is the pattern I followed with other functions in this PR.

Reply:
Resolving this thread based on your approval via Discord.